src/util/requests.js

```diff
@@ -3,10 +3,10 @@ const API_URL = process.env.REACT_APP_API_URL;
 // encodeURIComponent does not convert all URI-unfriendly characters, necessitating
 // and enhanced encoding function
 // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent#Return_value
-export const sanitizeKey = str => (
-  encodeURIComponent(str)
-    .replace(/[!'.()*]/g, ch => `%${ch.charCodeAt(0).toString(16)}`)
-);
+export const sanitizeKey = (str) => {
+  const getCharCode = ch => `%${ch.charCodeAt(0).toString(16).toUpperCase()}`;
+  return encodeURIComponent(str).replace(/[!'()*]/g, ch => getCharCode(ch));
```

Comment:
You should be able to supply `getCharCode` as the second argument to `replace`, so
```js
return encodeURIComponent(str).replace(/[!'()*]/g, getCharCode);
```